src/components/Modal/useModal.tsx

```diff
@@ -51,6 +52,7 @@ export function useModal(): UseModalHook {
           modalState.current.api.current.setSize(size);
         }
       },
+      inModal: () => isDefined(modalState.current),
```

Comment:
This could be an issue as anything within the BeamContext will think it is "inModal" when the modal is opened, even if it is behind the modal.